### ISSUE TITLE
iPad orientation error when bringing up share window from popover

### ISSUE THREAD

**Issue by ghost** [ideashower/ShareKit#113]
Hi,
If you are on an iPad and bring up a modal dialogue through ShareKit (such as through sharing an email), and then rotate the iPad before dismissing it, there is a problem where the popover is completely messed up afterwards for the lifetime of the app, having rotated the sub-contents of the popover. I believe this issue can be solved by enabling the modalInPopover setting on the view: http://efreedom.com/Question/1-3484987/Presenting-Modal-View-Controller-Popover. Perhaps you could add in a next version a setting for "willAppearFromPopup" or similar and set this flag properly?

**Comment by leearmstrong** [comment]
Is there a workaround that I can use today then as I have this issue also?

**Comment by ghost** [comment]
Yes, the workaround seems to be to set the root view controller to a higher level controller, and not the popup controller:

```
[[SHK currentHelper] setRootViewController:<yourcontroller>];
```

**Comment by leearmstrong** [comment]
Perfect thanks I notice that the HUD is then wrong way round.

**Comment by geerlingguy** [comment]
Subscribe - ran into this as well.
